src/components/OnboardingComponents/SemanticSearch/SemanticSearchContainer.tsx

```diff
@@ -0,0 +1,152 @@
+  const searchStatus = useAppSelector((state) => state.semanticSearch.searchStatus);
+  const {t} = useTranslation('global');
+  const [selectedArticle, setSelectedArticle] = useState<SearchResult | null>(null);
+
+  const suggestions = [t('semanticSearch.suggestions.scenarioForecasting'), t('semanticSearch.suggestions.purpose')];
+
+  const handleSearch = () => {
+    // performSearch is now required and will not be undefined.
+    if (performSearch) {
+      performSearch(query);
+    }
+  };
+
+  const handleClear = () => {
+    dispatch(setSemanticSearchQuery(''));
+    clearSearch();
+  };
+
+  const handleSuggestionClick = (suggestion: string) => {
+    dispatch(setSemanticSearchQuery(suggestion));
+    if (performSearch) {
+      performSearch(suggestion);
+    }
+  };
+
+  const handleResultClick = (article: SearchResult) => {
+    setSelectedArticle(article);
+  };
+
+  const handleCloseDialog = () => {
+    setSelectedArticle(null);
+  };
+
+  return (
+    <Box mt={4}>
+      <Divider sx={{mb: 3}} />
+      <Typography variant='h2' gutterBottom>
+        {t('semanticSearch.title')}
+      </Typography>
+      <Typography variant='body1' paragraph sx={{color: 'GrayText'}}>
+        {t('semanticSearch.description')}
+      </Typography>
+
+      <TextField
+        fullWidth
+        size='small'
+        variant='outlined'
+        placeholder={t('semanticSearch.placeholder')}
+        value={query}
+        onChange={(e) => dispatch(setSemanticSearchQuery(e.target.value))}
+        onKeyDown={(e) => {
+          if (e.key === 'Enter') {
+            handleSearch();
+          } else if (e.key === 'Escape') {
+            handleClear();
+          }
+        }}
+        InputProps={{
+          sx: {
+            paddingRight: '4px',
+          },
+          endAdornment: (
+            <InputAdornment position='end'>
+              <IconButton
+                onClick={handleSearch}
+                disabled={isLoading}
+                sx={{
+                  width: '32px',
+                  height: '32px',
+                  borderRadius: '4px',
+                  backgroundColor: 'primary.main',
+                  color: 'primary.contrastText',
+                  '&:hover': {
+                    backgroundColor: 'primary.dark',
+                  },
+                  '&.Mui-disabled': {
+                    backgroundColor: 'action.disabledBackground',
+                  },
+                }}
+              >
+                {isLoading ? <CircularProgress size={24} color='inherit' /> : <SearchIcon sx={{fontSize: '16px'}} />}
+              </IconButton>
+            </InputAdornment>
+          ),
+        }}
+      />
+
+      <QuerySuggestions suggestions={suggestions} onSuggestionClick={handleSuggestionClick} />
+
+      <Box
+        mt={3}
+        sx={{
+          maxHeight: '450px',
+          overflowY: 'auto',
+          p: '2px',
+        }}
```

Comment:
same here. Especially don't mix different styling methods.
```suggestion
      <Box
        sx={{
          marginTop: 3,
          maxHeight: '450px',
          overflowY: 'auto',
          padding: '2px',
        }}
```

Reply:
Done